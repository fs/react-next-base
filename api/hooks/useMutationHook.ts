import type { DocumentNode } from 'graphql';
import type { TypedDocumentNode } from '@graphql-typed-document-node/core';
import type {
  MutationTuple,
  MutationHookOptions,
  FetchResult,
  MutationOptions,
  ApolloClient,
  InMemoryCache,
} from '@apollo/client';

import { useMutation } from '@apollo/client';

export const mutationApi = <TData, TVariables>(
  apolloClient: ApolloClient<InMemoryCache>,
  options: MutationOptions<TData, TVariables>,
): Promise<FetchResult<TData>> => {
  return apolloClient.mutate(options);
};

const useMutationHook = <TData, TVariables>(
  mutation: DocumentNode | TypedDocumentNode<TData, TVariables>,
  options?: MutationHookOptions<TData, TVariables> | undefined,
): MutationTuple<TData, TVariables> => {
  return useMutation(mutation, options);
};

export default useMutationHook;
