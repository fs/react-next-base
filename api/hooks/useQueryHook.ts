import type { DocumentNode } from 'graphql';
import type { TypedDocumentNode } from '@graphql-typed-document-node/core';
import type { QueryHookOptions, QueryResult } from '@apollo/client';

import { useQuery } from '@apollo/client';

const useQueryHook = <TData, TVariables>(
  query: DocumentNode | TypedDocumentNode<TData, TVariables>,
  options?: QueryHookOptions<TData, TVariables> | undefined,
): QueryResult<TData, TVariables> => {
  return useQuery(query, options);
};

export default useQueryHook;
