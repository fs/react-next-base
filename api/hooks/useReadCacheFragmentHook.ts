import type { DataProxy, ApolloClient, InMemoryCache } from '@apollo/client';

export const readCacheFragmentApi = <TData, TVariables>(
  apolloClient: ApolloClient<InMemoryCache>,
  options: DataProxy.Fragment<TVariables, TData>,
  optimistic?: boolean,
): TData | null => {
  return apolloClient.readFragment(options, optimistic);
};
