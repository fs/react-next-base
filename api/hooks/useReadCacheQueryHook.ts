import type { DataProxy, ApolloClient, InMemoryCache } from '@apollo/client';

export const readCacheQueryApi = <TData, TVariables>(
  apolloClient: ApolloClient<InMemoryCache>,
  options: DataProxy.Query<TVariables, TData>,
  optimistic?: boolean,
): TData | null => {
  return apolloClient.readQuery(options, optimistic);
};
