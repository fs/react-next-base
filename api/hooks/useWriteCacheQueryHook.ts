import type { ApolloCache, Cache, Reference, DataProxy, ApolloClient, InMemoryCache } from '@apollo/client';

export const writeCacheQueryApi = <TData, TVariables>(
  apolloClient: ApolloClient<InMemoryCache>,
  options: DataProxy.WriteQueryOptions<TData, TVariables>,
): void => {
  return apolloClient.writeQuery(options);
};

const useWriteCacheQueryHook = <TResponseData, TData, TVariables>(
  cache: ApolloCache<TResponseData>,
  options: Cache.WriteQueryOptions<TData, TVariables>,
): Reference | undefined => {
  return cache.writeQuery(options);
};

export default useWriteCacheQueryHook;
