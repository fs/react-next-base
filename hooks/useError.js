import { useContext } from 'react';
import ErrorContext from 'contexts/ErrorContext';

const useError = () => {
  const { error, setError } = useContext(ErrorContext);
  return { error, setError };
};

export default useError;
