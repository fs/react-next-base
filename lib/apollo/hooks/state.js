import { useQuery } from '@apollo/client';
import CurrentUser from 'graphql/queries/currentUser.graphql';

export const useCurrentUser = (prefetch = true) => {
  const { loading, data, error } = useQuery(CurrentUser, {
    fetchPolicy: prefetch ? 'cache-first' : 'cache-only',
  });

  const user = data?.me;

  return { user, loading, error };
};
