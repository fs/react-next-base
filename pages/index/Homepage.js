import React from 'react';
import styled from 'styled-components';
import DefaultTemplate from 'components/templates/DefaultTemplate';

const Title = styled.h1`
  font-size: 2rem;
  text-align: center;
`;

const Home = () => (
  <DefaultTemplate>
    <Title>Welcome to React Next Base</Title>
  </DefaultTemplate>
);

export default Home;
