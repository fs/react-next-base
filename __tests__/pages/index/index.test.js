import React from 'react';
import renderWithApolloClient from '__tests__/helpers/renderWithApolloClient';
import { render, screen } from '@testing-library/react';
import 'jest-styled-components';
import renderWithTheme from '__tests__/helpers/renderWithTheme';
import { useCurrentUser } from 'lib/apollo/hooks/state';

import Homepage from 'pages/index';

jest.mock('lib/apollo/hooks/state.js');

describe('Homepage', () => {
  test('should render correctly', async () => {
    // Arrange
    const mockUseCurrentUser = jest.fn(() => ({
      loading: undefined,
      error: undefined,
      user: { id: '1', email: 'user@mail.ru' },
    }));
    useCurrentUser.mockImplementation(mockUseCurrentUser);

    // Act
    render(renderWithTheme(renderWithApolloClient(<Homepage />)));
    const container = await screen.getByTestId('page-content');

    // Assert
    expect(container).toMatchSnapshot();
  });
});
