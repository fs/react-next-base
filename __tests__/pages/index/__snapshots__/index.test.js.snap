// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage should render correctly 1`] = `
.c0 {
  font-size: 2rem;
  text-align: center;
}

@media (max-width:576px) {

}

@media (max-width:1200px) {

}

<div
  class=""
  data-testid="test-page-content"
>
  <h1
    class="c0 capitalize"
    data-cy="welcome-page"
  >
    Welcome to React Next Base
  </h1>
</div>
`;
